mpc: Update to 0.30. Use autotools

thess opened the pull request:
Maintainer: me 
Compile tested: MIPS_24kc
Run tested: AR71xx/WNDR4300

Posted for review and to give folks a chance to reject this somewhat gross hack. I'm totally open to simpler alternatives to moving forward with packages needing the python3/meson/ninja toolchain and configurations for cross compiling. 

Description:

Package versions newer than 0.28 require meson/ninja which OpenWrt does not
    support in its build environment. These files provide the minimum autotool
    configuration necessary to build later versions.

Signed-off-by: Ted Hess <thess@kitschensync.net>

diizzyy commented:
@thess 
Travis doesn't seem happy :-/

thess commented:
Travis doesn't seem to have AM_LANGINFO_CODESET in the library (intl.m4 on my system). I'm going to check out a buildbot I have access to tomorrow and see if it builds on Debian 8. Other choice is to remove the dependency since it isn't executed anyway when --disable-iconv is set. Travis is marginal at best.

neheb commented:
Seems like a good solution for now.

Why the need for std=gnu99? GCC defaults to gnu11.

thess commented:
@neheb - Regarding use of gnu99... Good question. It's a carryover from their build specs. Anyway, I removed it with no ill effects. Also removed the AM_LANGINFO_CODESET check from _configure.ac_
